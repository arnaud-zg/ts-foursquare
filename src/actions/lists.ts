import { NRequest } from '../../types'
import { getObservableLists } from '../services/lists'
import { IStandaloneConfig } from '../standalone'

export interface GetListsProps {
  config?: IStandaloneConfig
  payload: NRequest.IListsPayload
}

export const getLists = ({ config, payload }: GetListsProps) =>
  getObservableLists({ config, payload }).toPromise()
