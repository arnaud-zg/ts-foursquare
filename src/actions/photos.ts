import { NRequest } from '../../types'
import { getObservablePhotosDetails } from '../services/photos'
import { IStandaloneConfig } from '../standalone'

export interface GetPhotosDetailsProps {
  config?: IStandaloneConfig
  payload: NRequest.IPhotosDetailsPayload
}

export const getPhotosDetails = ({ config, payload }: GetPhotosDetailsProps) =>
  getObservablePhotosDetails({ config, payload }).toPromise()
