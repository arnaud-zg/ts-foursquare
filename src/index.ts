require('isomorphic-fetch')

export * from './standalone'
export * from './utils'

export * from './DEPRECATED_actions'
export * from './DEPRECATED_reducers'
export * from './DEPRECATED_selectors'
