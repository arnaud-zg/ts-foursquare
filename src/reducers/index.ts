import { NVenues } from '../../types/venues'

export type TRootState = {
  venues: NVenues.IVenuesState
}
