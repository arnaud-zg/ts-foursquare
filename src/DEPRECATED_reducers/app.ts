import { combineReducers } from 'redux'
import { initialState as initialStateLife, lifeReducer } from './life'
import { initialState as initialStateLists, listsReducer } from './lists'
import { initialState as initialStatePhotos, photosReducer } from './photos'
import { initialState as initialStateRequest, requestReducer } from './request'
import { initialState as initialStateStatus, statusReducer } from './status'
import { initialState as initialStateVenues, venuesReducer } from './venues'

export const initialState = {
  life: initialStateLife,
  lists: initialStateLists,
  photos: initialStatePhotos,
  request: initialStateRequest,
  status: initialStateStatus,
  venues: initialStateVenues,
}

export type TState = typeof initialState

export const appReducer = combineReducers({
  life: lifeReducer,
  lists: listsReducer,
  photos: photosReducer,
  request: requestReducer,
  status: statusReducer,
  venues: venuesReducer,
})
