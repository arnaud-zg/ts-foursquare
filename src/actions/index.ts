import { TLifeAction } from './life'
import { TVenuesAction } from './venues'

export type TRootActionType = TVenuesAction | TLifeAction

export type TRootAction = TVenuesAction | TLifeAction

export * from './life'
export * from './venues'
