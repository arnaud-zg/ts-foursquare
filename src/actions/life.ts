import { NLife } from '../../types/life'
import { lifeReducer } from 'reducers/life'

export enum EActionRequest {
  GET_LIFE = 'GET_LIFE',
}

export const getLife = (): Promise<NLife.ILife> =>
  new Promise(resolve => {
    resolve(
      lifeReducer({
        type: EActionRequest.GET_LIFE,
        payload: {
          life: true,
        },
      })
    )
  })
