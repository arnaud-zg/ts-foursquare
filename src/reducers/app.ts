import { combineReducers } from 'redux'
import { initialState as initialStateLife } from '../../src/reducers/life'
import { initialState as initialStateStatus } from '../../src/reducers/status'
import { initialState as initialStateVenues } from '../../src/reducers/venues'
import { lifeReducer } from './life'
import { statusReducer } from './status'
import { venuesReducer } from './venues'

export const initialState = {
  life: initialStateLife,
  status: initialStateStatus,
  venues: initialStateVenues,
}

export type TState = typeof initialState

export const appReducer = combineReducers({
  life: lifeReducer,
  status: statusReducer,
  venues: venuesReducer,
})
