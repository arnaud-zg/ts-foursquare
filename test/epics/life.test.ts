import { Action } from 'typesafe-actions'
import { getAccessTokenAsync } from '../../src/actions'
import { adaptGetAccessToken } from '../../src/adapter'
import { getAccessTokenEpic } from '../../src/epics'
import { initialState } from '../../src/reducers/app'
import {
  mockingFailingNotFoundFetch,
  mockingFailingNotValidFetch,
  mockingFetch,
  testEpic,
} from '../../src/utils/test'

require('isomorphic-fetch')

describe('epics/life', () => {
  test.each`
    scenario                     | action                                           | payload             | expectedActions                                                                 | epic
    ${'should get access token'} | ${getAccessTokenAsync.request({ code: '1234' })} | ${{ code: '5678' }} | ${[getAccessTokenAsync.success(adaptGetAccessToken({ access_token: '5678' }))]} | ${getAccessTokenEpic}
  `('$scenario', ({ action, expectedActions, epic, payload, done }) => {
    mockingFetch({ response: { ...payload } })
    testEpic(
      epic,
      expectedActions.length,
      action,
      (actions: Action[]) => {
        expect(actions).toMatchSnapshot()
        actions.forEach((actualAction, index) => {
          expect(actualAction).toEqual(expectedActions[index])
        })
        done()
      },
      initialState
    )
  })
})

describe('epics/lists | error case: no network', () => {
  beforeAll(() => {
    mockingFailingNotFoundFetch()
  })

  test.each`
    scenario                     | action                                           | epic
    ${'should get access token'} | ${getAccessTokenAsync.request({ code: '1234' })} | ${getAccessTokenEpic}
  `('$scenario', ({ action, epic, done }) => {
    testEpic(
      epic,
      1,
      action,
      (actions: Action[]) => {
        expect(actions).toMatchSnapshot()
        done()
      },
      initialState
    )
  })
})

describe('epics/venues | error case: not valid response', () => {
  beforeAll(() => {
    mockingFailingNotValidFetch()
  })

  test.each`
    scenario                     | action                                           | epic
    ${'should get access token'} | ${getAccessTokenAsync.request({ code: '1234' })} | ${getAccessTokenEpic}
  `('$scenario', ({ action, epic, done }) => {
    testEpic(
      epic,
      1,
      action,
      (actions: Action[]) => {
        expect(actions).toMatchSnapshot()
        done()
      },
      initialState
    )
  })
})
