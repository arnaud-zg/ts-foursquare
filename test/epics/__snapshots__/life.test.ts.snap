// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`epics/life should get access token 1`] = `
Array [
  Object {
    "meta": 0,
    "payload": Object {
      "accessToken": undefined,
    },
    "type": "GET_ACCESS_TOKEN_SUCCESS",
  },
]
`;

exports[`epics/lists | error case: no network should get access token 1`] = `
Array [
  Object {
    "payload": [Error: Error 404],
    "type": "GET_ACCESS_TOKEN_FAILURE",
  },
]
`;

exports[`epics/venues | error case: not valid response should get access token 1`] = `
Array [
  Object {
    "payload": [Error: response.json is not a function],
    "type": "GET_ACCESS_TOKEN_FAILURE",
  },
]
`;
